lib/spack/spack/reporters/cdash.py

```diff
@@ -51,39 +55,57 @@ class CDash(Reporter):
     CDash instance hosted at https://mydomain.com/cdash.
     """
 
-    def __init__(self, install_command, cdash_upload_url):
-        Reporter.__init__(self, install_command, cdash_upload_url)
+    def __init__(self, args):
+        Reporter.__init__(self, args)
         self.template_dir = os.path.join('reports', 'cdash')
-        self.hostname = socket.gethostname()
+        self.cdash_upload_url = args.cdash_upload_url
+        self.install_command = ' '.join(args.package)
+        if args.cdash_build:
+            self.buildname = args.cdash_build
+        else:
+            self.buildname = self.install_command
```

Comment:
You can write this conditional in a more idiomatic way like:
```python
self.buildname = args.cdash_build or self.install_command
```

Reply:
Thanks for the tip! I agree this looks much cleaner. I'll push this change shortly.